sdk/storage/Azure.Storage.DataMovement.Blobs/src/DataMovementBlobsExtensions.cs

```diff
@@ -445,5 +449,233 @@ internal static CommitBlockListOptions ToCommitBlockOptions(this BlockBlobStorag
+                    transferId,
+                    startIndex,
+                    cancellationToken: cancellationToken).ConfigureAwait(false);
+                options.AccessTier = accessTier.ToAccessTier();
+            }
+            return options;
+        }
+
+        internal static async Task<BlobStorageResourceOptions> GetBlobResourceOptionsAsync(
+            this TransferCheckpointer checkpointer,
+            string transferId,
+            bool isSource,
+            CancellationToken cancellationToken)
+        {
+            BlobStorageResourceOptions options = new BlobStorageResourceOptions();
+
+            // TODO: parse out the rest of the parameters from the Job Part Plan File
+
+            if (!isSource)
+            {
+                // Get Metadata
+                int metadataIndex = DataMovementConstants.PlanFile.DstBlobMetadataLengthIndex;
+                int metadataReadLength = DataMovementConstants.PlanFile.DstBlobTagsLengthIndex - metadataIndex;
+                string metadata = await GetHeaderValue(
+                    checkpointer,
+                    transferId,
+                    metadataIndex,
+                    metadataReadLength,
+                    DataMovementConstants.PlanFile.MetadataStrMaxLength,
+                    cancellationToken).ConfigureAwait(false);
+                options.Metadata = metadata.ToDictionary(nameof(metadata));
+
+                // Get blob tags
+                int tagsIndex = DataMovementConstants.PlanFile.DstBlobTagsLengthIndex;
+                int tagsReadLength = DataMovementConstants.PlanFile.DstBlobIsSourceEncrypted - tagsIndex;
+                string tags = await GetHeaderValue(
+                    checkpointer,
+                    transferId,
+                    tagsIndex,
+                    tagsReadLength,
+                    DataMovementConstants.PlanFile.BlobTagsStrMaxLength,
+                    cancellationToken).ConfigureAwait(false);
+                options.Tags = tags.ToDictionary(nameof(tags));
+            }
+            return options;
+        }
+
+        internal static async Task<BlobStorageResourceContainerOptions> GetBlobContainerOptionsAsync(
+            this TransferCheckpointer checkpointer,
+            string directoryPrefix,
+            string transferId,
+            bool isSource,
+            CancellationToken cancellationToken)
+        {
+            BlobStorageResourceOptions baseOptions = await checkpointer.GetBlobResourceOptionsAsync(
+                transferId,
+                isSource,
+                cancellationToken).ConfigureAwait(false);
+            BlobStorageResourceContainerOptions options = new()
+            {
+                DirectoryPrefix = directoryPrefix,
+                ResourceOptions = baseOptions,
+            };
+
+            return options;
+        }
+
+        private static async Task<string> GetHeaderValue(
```

Comment:
Should GetHeaderValue and GetByteValue go into the checkpointer itself rather then be extension methods in the Blob package. I ended up needing something similar for GetResumableTransfers in the base package so I think these might make sense as checkpointer methods?

Reply:
While I do agree it should probably live in the checkpointer, in the future the way we grab each header value may change if the schema format changes. So we may only use this method in the soon to be older schemas.

Reply:
Moved this to the base extensions.